### ISSUE TITLE
Use MAUI essentials in a Console app

### ISSUE THREAD

**Issue by profix898** [dotnet/maui#17317]
### Description

I'm trying to use MAUI essentials in a non-MAUI project with .NET 8 preview 7.

I created a simple console application targeting net8.0-windows (platform specific TFM). `UseMauiEssentials` is set in CSPROJ and I have also included a package reference to *Microsoft.Maui.Essentials* (8.0.0-preview.7.8842), which according to a build message is now required.

The project compiles fine. But trying to use a MAUI essentials API - here `FileSystem.AppDataDirectory` - raises the following exception:

`
Microsoft.Maui.ApplicationModel.NotImplementedInReferenceAssemblyException: 'This functionality is not implemented in the portable version of this assembly. You should reference the NuGet package from your main application project in order to reference the platform-specific implementation.'
`

I'm on VS 17.8.0 Preview 1.0 with .NET 8.0.100-preview.7.23376.3 installed. The .NET MAUI SDK is checked in the VS installer, but `dotnet workload list` does not list anything. A normal/full MAUI app works fine though. I also tried to manually install workloads via `dotnet workload install maui`, but the scenario above still does not work.

### Link to public reproduction project repository

https://github.com/profix898/EssentialsApp1

### Version with bug

8.0.0-preview.7.8842

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Relevant log output

```shell
> dotnet --list-sdks                                                                                          
3.1.426 [C:\Program Files\dotnet\sdk]
6.0.316 [C:\Program Files\dotnet\sdk]
8.0.100-preview.7.23376.3 [C:\Program Files\dotnet\sdk]
```

**Comment by mattleibow** [comment]
I had a look at this and it appears the issue is with the Windows App SDK. As soon as you add that package it breaks the build.

Maui uses the Windows App SDK for Windows apps and to do Windows things. I will investigate why this fails in a console app. However, .NET MAUI and components are designed to be run within a Windows App SDK / WinUI 3 app. I tested that and it works in .NET 8.

I did have to add a property in RC1 due to a very late change `<UseRidGraph>true</UseRidGraph>`

However, you can make a Windows App SDK app also behave as a console app, with the exception that it does not support Any CPU and instead needs a x64/x86/arm64.

This is my csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

	<PropertyGroup>
		<OutputType>Exe</OutputType>
		<TargetFramework>net8.0-windows10.0.19041.0</TargetFramework>
		<TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
		<RootNamespace>App34</RootNamespace>
		
		<!-- Needed for Windows App SDK as they don't support Any CPU -->
		<Platforms>x86;x64;ARM64</Platforms>
		<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>

		<!-- Needed for .NET RC 1 -->
		<UseRidGraph>true</UseRidGraph>

		<!-- Needed to enable modern Windows App SDK projects -->
		<EnableMsixTooling>true</EnableMsixTooling>
		<WindowsPackageType>None</WindowsPackageType>
		<WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained>
	</PropertyGroup>

	<ItemGroup>
		<PackageReference Include="Microsoft.Maui.Essentials" Version="8.0.0-rc.1.9171" />
	</ItemGroup>

</Project>
```

**Comment by profix898** [comment]
Thanks at lot @mattleibow. I have added your suggested changes to my test project and it just works (even for a console app, targeting x86/x64/arm64) with .NET 8 RC1.

**Note:** Console app was merely meant as an example. I think the MAUI team should realize that Essentials is very useful outside the MAUI UI world. Everyone doing cross-platform work with other UI frameworks would greatly benefit from a more prominent and broader support (potentially even taking contributions to expand platform reach, e.g. Linux, Tizen, etc.) of the package. Its great to see that with .NET 8 we can (again) simply include 'Microsoft.Maui.Essentials' from NuGet (given a correct CSProj configuration of the projects).
It would be great to have this documented somewhere, so others looking for Essentials outside MAUI UI can follow along.

**Comment by mattleibow** [comment]
This is only an issue with WinUI at the moment, but essentials should work with iOS, android and Mac catalyst with no problem. You should be able to use this with any UI framework, plain android or even uno framework. Windows is a bit annoying as there are a few extra steps.

Let me know if other platforms have issues.

And, we are willing to take PRs to add platforms that we are missing. I think we should also port the macos work we had in xamarin.essentials...

The issue more is the plain console apps... They tend not to have an api that is easily usable. Like win32 will be a bunch of pinvokes. But, if there is a platform that you are needing, maybe open a new issue now that I have commandeered this issue for console apps.

I will reopen this one so we can have an item tracking this for said console apps.

**Comment by orosbogdan** [comment]
Yes this is very much needed for cross platform apps that don't use MAUI.
